### DISCUSSION TITLE
Add parameter names to functions and type constructors `package_interface.json`

### DISCUSSION THREAD

**Discussion by just-maiyak** [gleam-lang/gleam#3462] [Ideas & suggestions]
I am exploring the `package_interface.json` spec in an effort, with @GearsDatapacks, to build a CLI tool to peek at package docs (somewhat related to https://github.com/gleam-lang/gleam/issues/2288).
I think the only thing missing to print out full function signatures would be the symbols for function parameters. I see labels and ids are already present in the spec, but I figure that the signature would be hard to read if there are no labels. I also see that the function generating the `package_interface.json` [is already passed the AST](https://github.com/gleam-lang/gleam/blob/3128e3bfaf84d9986aacf8d4a784ad6fe9df8c6f/compiler-cli/src/docs.rs#L142), so i figure it is possible to add the information in the file.
Could parameter names be added in the spec of `package_interface.json` ?

**Comment by lpil** [comment]
Parameter names are internal and  not part of the public interface of a function.

**Reply by just-maiyak** [reply]
Okay, we'll just print the types then. Thanks 🙂
